cbc-repeated-iv/src/cbc_game.rs

```diff
@@ -0,0 +1,397 @@
+    key: [u8; 16],
+    iv: [u8; 16],
+
+    known_plaintext: Vec<u8>,
+    known_ciphertext: Vec<u8>,
+
+    plaintexts_vec: Vec<Vec<u8>>,
+    ciphertexts_vec: Vec<Vec<u8>>,
+
+    cbc_same_iv_ascii_art: String,
+}
+
+impl Default for CBCGame {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CBCGame {
+    pub fn new() -> Self {
+        Self {
+            key: [0; 16],
+            iv: [0; 16],
+
+            known_plaintext: br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#
+                .to_vec(),
+            known_ciphertext: vec![],
+
+            plaintexts_vec: vec![
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "M"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Tom",  "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "Administrator", "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{ "Heartbeat": "Alive",         "Since": "1653924357" }"#.to_vec(),
+            ],
+            ciphertexts_vec: vec![],
+
+            cbc_same_iv_ascii_art: "
+
+     ▄████████ ▀█████████▄   ▄████████         ▄████████    ▄████████   ▄▄▄▄███▄▄▄▄      ▄████████       ▄█   ▄█    █▄
+    ███    ███   ███    ███ ███    ███        ███    ███   ███    ███ ▄██▀▀▀███▀▀▀██▄   ███    ███      ███  ███    ███
+    ███    █▀    ███    ███ ███    █▀         ███    █▀    ███    ███ ███   ███   ███   ███    █▀       ███▌ ███    ███
+    ███         ▄███▄▄▄██▀  ███               ███          ███    ███ ███   ███   ███  ▄███▄▄▄          ███▌ ███    ███
+    ███        ▀▀███▀▀▀██▄  ███             ▀███████████ ▀███████████ ███   ███   ███ ▀▀███▀▀▀          ███▌ ███    ███
+    ███    █▄    ███    ██▄ ███    █▄                ███   ███    ███ ███   ███   ███   ███    █▄       ███  ███    ███
+    ███    ███   ███    ███ ███    ███         ▄█    ███   ███    ███ ███   ███   ███   ███    ███      ███  ███    ███
+    ████████▀  ▄█████████▀  ████████▀        ▄████████▀    ███    █▀   ▀█   ███   █▀    ██████████      █▀    ▀██████▀
+
+".to_string()
+        }
+    }
+
+    pub fn start(&mut self) -> anyhow::Result<()> {
+        let mut iter = 0;
+
+        println!("{}", self.cbc_same_iv_ascii_art.on_color(Color::Green));
+
+        self.reinitialize_enc_dec();
+
+        self.encrypt_all_texts();
+
+        self.print_known_texts();
+
+        while iter < self.ciphertexts_vec.len() {
+            let mut line = String::new();
+            println!(
+                "\n{}",
+                "❱".repeat(40).color(Color::Black).on_color(Color::Yellow)
+            );
+            println!("Input key (c(continue)/q(quit)):");
+            std::io::stdin().read_line(&mut line)?;
+            // remove new line char if present and leading and tracing spaces
+            line = line.trim().to_string();
+            match line.to_lowercase().as_str() {
```

Comment:
These two lines contain two allocations: first in `to_string`, and second in `to_lowercase`.  It could be eliminated to only one:
```suggestion
            match line.trim().to_lowercase().as_str() {
```